### PR TITLE
Adds a utility function to transform AutoCompleteMatch

### PR DESCRIPTION
**Pull request by elliette**
The new utility function is re-used in the `file_search_test` to capitalize matching segments, making it a lot easier to verify that the output is expected. 

Work towards: https://github.com/flutter/devtools/issues/3554

### CODE REVIEW

File: packages/devtools_app/lib/src/ui/search.dart [hunk 1]
```diff
@@ -1256,4 +1221,36 @@ class AutoCompleteMatch {
 
   final String text;
   final List<Range> matchedSegments;
+
+  // Transform the autocomplete match somehow (e.g. create a TextSpan where the
+  // matched segments are highlighted).
+  T transformAutoCompleteMatch<T>(
+      {T transformMatchedSegment(String segment),
+      T transformUnmatchedSegment(String segment),
+      T combineSegments(List<T> segments)}) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/ui/search.dart:1230]: nit: trailing comma

File: packages/devtools_app/lib/src/ui/search.dart [hunk 2]
```diff
@@ -380,55 +380,20 @@ class AutoCompleteState extends State<AutoComplete> with AutoDisposeMixin {
...
-      );
-    }
-
-    return TextSpan(
-      text: spans.first.text,
-      style: spans.first.style,
-      children: spans.sublist(1),
+      ),
+      combineSegments: (segments) => TextSpan(
+        text: segments.first.text,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/ui/search.dart:393]: We should have some length checks on `segments`. `.first` and `.sublist` will throw exceptions if the list is empty

File: packages/devtools_app/lib/src/ui/search.dart [hunk 3]
```diff
@@ -1256,4 +1221,36 @@ class AutoCompleteMatch {
 
   final String text;
   final List<Range> matchedSegments;
+
+  // Transform the autocomplete match somehow (e.g. create a TextSpan where the
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/ui/search.dart:1225]: use /// for dart doc

**Review by kenzieschmoll — APPROVED** [review]

File: packages/devtools_app/lib/src/ui/search.dart (hunk 1 above)
Comment (elliette) [reply packages/devtools_app/lib/src/ui/search.dart]: Good catch, done!

File: packages/devtools_app/lib/src/ui/search.dart (hunk 2 above)
Comment (elliette) [reply packages/devtools_app/lib/src/ui/search.dart:393]: Good point, added an `assert` in the `transformAutocompleteMatch` method to verify segments is not empty.

File: packages/devtools_app/lib/src/ui/search.dart (hunk 3 above)
Comment (elliette) [reply packages/devtools_app/lib/src/ui/search.dart]: Done, thanks!

**Pull request merged**